crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -397,38 +370,40 @@ impl<'src> Parser<'src> {
         }
     }
 
+    fn take_value(&mut self, kind: TokenKind) -> TokenValue {
+        let value = self.tokens.take_value();
+        self.bump(kind);
+        value
+    }
+
     /// Bumps the current token assuming it is of the given kind.
     ///
-    /// Returns the current token as an owned value.
-    ///
     /// # Panics
     ///
     /// If the current token is not of the given kind.
-    fn bump(&mut self, kind: TokenKind) -> (Tok, TextRange) {
+    fn bump(&mut self, kind: TokenKind) {
         assert_eq!(self.current_token_kind(), kind);
 
-        self.next_token()
+        self.next_token();
     }
 
     /// Bumps the current token assuming it is found in the given token set.
     ///
-    /// Returns the current token as an owned value.
-    ///
     /// # Panics
     ///
     /// If the current token is not found in the given token set.
-    fn bump_ts(&mut self, ts: TokenSet) -> (Tok, TextRange) {
+    fn bump_ts(&mut self, ts: TokenSet) {
         assert!(ts.contains(self.current_token_kind()));
 
-        self.next_token()
+        self.next_token();
     }
 
     fn expect(&mut self, expected: TokenKind) -> bool {
         if self.eat(expected) {
             return true;
         }
 
-        let (found, range) = self.current_token();
+        let (found, range) = (self.current_token_kind(), self.current_token_range());
         self.add_error(ParseErrorType::ExpectedToken { found, expected }, range);
```

Comment:
It looks a bit strange to create a tuple here just to destruct again. I think I would inline the expression now.

```suggestion
        let (found, range) = (self.current_token_kind(), self.current_token_range());
        self.add_error(
        	ParseErrorType::ExpectedToken { found: self.current_token_kind(), expected },
        	self.current_token_range()
      	);
```